motr/m0crate/crate.c

```diff
@@ -1254,6 +1257,32 @@ static void cr_btree_delete(struct m0_key_val *kv)
 			  bo_u.u_btree.t_rc);
 }
 
+static void cr_btree_warmup(struct cr_workload_btree *cwb)
+{
+	char 			v[cwb->cwb_max_val_size];
+	struct cr_btree_key 	cbk;
+
+	int i		= 0;
+	int count	= cwb->cwb_warmup_put_cnt;
+	int ksize	= cwb->cwb_key_size;
+	int vsize	= cwb->cwb_val_size != -1 ? cwb->cwb_val_size :
+			  getrnd(1, cwb->cwb_max_val_size);
+	int pattern	= cwb->cwb_pattern;
+	uint64_t k 	= cwb->cwb_bo[BOT_INSERT].key;
+
+	for (i = 0; i < count ; i++)
+	{
+		struct m0_key_val dummy_kv;
+
+		cr_btree_key_make(ksize, k, pattern, &cbk);
```

Comment:
instead of k pass cwb->cwb_bo[BOT_INSERT].key to cr_btree_key_make and then increment cwb->cwb_bo[BOT_INSERT].key. This will make sure that on mix workloads (for e.g. insert +lookup), inserts have head start by warm up put count.